packages/devtools_app/lib/src/screens/vm_developer/vm_field_display.dart

```diff
@@ -0,0 +1,186 @@
+          child: Column(
+            children: [
+              FieldInfoWidget(
+                fieldDataRows: _fieldDataRows(field),
+              ),
+              Flexible(
+                child: ListView(
+                  children: [
+                    RetainingPathWidget(
+                      retainingPath: field.retainingPath,
+                      onExpanded: _onExpandRetainingPath,
+                    ),
+                    InboundReferencesWidget(
+                      inboundReferences: field.inboundReferences,
+                      onExpanded: _onExpandInboundRefs,
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+
+  void _onExpandRetainingPath(bool expanded) {
+    if (field.retainingPath.value == null) field.requestRetainingPath();
+  }
+
+  void _onExpandInboundRefs(bool expanded) {
+    if (field.inboundReferences.value == null) field.requestInboundsRefs();
+  }
+}
+
+/// Displays general VM information of the Class Object.
+class FieldInfoWidget extends StatelessWidget implements PreferredSizeWidget {
+  const FieldInfoWidget({
+    required this.fieldDataRows,
+  });
+
+  final List<MapEntry<String, Object?>> fieldDataRows;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox.fromSize(
+      size: preferredSize,
+      child: VMInfoCard(
+        title: 'General Information',
+        rowKeyValues: fieldDataRows,
+      ),
+    );
+  }
+
+  @override
+  Size get preferredSize => Size.fromHeight(
+        areaPaneHeaderHeight +
+            fieldDataRows.length * defaultRowHeight +
+            defaultSpacing,
+      );
+}
+
+/// Generates a list of key-value pairs (map entries) that will be displayed
+/// in the data rows of the FieldInfoWidget.
+List<MapEntry<String, Object?>> _fieldDataRows(FieldObject field) {
+  return [
+    MapEntry('Object Class', field.obj.type),
+    MapEntry(
+      'Shallow Size',
+      prettyPrintBytes(
+        field.obj.size ?? 0,
+        includeUnit: true,
+        kbFractionDigits: 1,
+        maxBytes: 512,
+      ),
+    ),
+    MapEntry(
+      'Reachable Size',
+      ValueListenableBuilder<bool>(
+        valueListenable: field.fetchingReachableSize,
+        builder: (context, fetching, _) => fetching
+            ? const CircularProgressIndicator()
+            : RequestableSizeWidget(
+                requestedSize: field.reachableSize,
+                requestFunction: field.requestReachableSize,
+              ),
+      ),
+    ),
+    MapEntry(
+      'Retained Size',
+      ValueListenableBuilder<bool>(
+        valueListenable: field.fetchingRetainedSize,
+        builder: (context, fetching, _) => fetching
+            ? const CircularProgressIndicator()
+            : RequestableSizeWidget(
+                requestedSize: field.retainedSize,
+                requestFunction: field.requestRetainedSize,
+              ),
+      ),
+    ),
```

Comment:
this code is all duplicated, too. If there are general information rows that will be there for any VmObject (or more than one), create a helper to generate these rows and use it here and in the class display code